Source/WebCore/dom/Node.cpp

```diff
@@ -2521,27 +2523,31 @@ void Node::defaultEventHandler(Event& event)
         }
 #endif
     } else if (eventType == eventNames.textInputEvent) {
-        if (is<TextEvent>(event)) {
+        if (RefPtr textEvent = dynamicDowncast<TextEvent>(event)) {
             if (RefPtr frame = document().frame())
-                frame->eventHandler().defaultTextInputEventHandler(downcast<TextEvent>(event));
+                frame->eventHandler().defaultTextInputEventHandler(*textEvent);
         }
 #if ENABLE(PAN_SCROLLING)
-    } else if (eventType == eventNames.mousedownEvent && is<MouseEvent>(event)) {
-        if (downcast<MouseEvent>(event).button() == MouseButton::Middle) {
+    } else if (auto* mouseEvent = dynamicDowncast<MouseEvent>(event); mouseEvent && eventType == eventNames.mousedownEvent) {
+        if (mouseEvent->button() == MouseButton::Middle) {
             if (enclosingLinkEventParentOrSelf())
                 return;
 
-            RenderObject* renderer = this->renderer();
-            while (renderer && (!is<RenderBox>(*renderer) || !downcast<RenderBox>(*renderer).canBeScrolledAndHasScrollableArea()))
-                renderer = renderer->parent();
-
-            if (renderer) {
+            RenderBox* renderBox = nullptr;
```

Comment:
Hm... I think the old code was easier to read.

Reply:
I will look into it. If you have a solution you prefer that relies on dynamic cast, I'd be happy to take it.
Personally, I felt that the new code was as clear, just less concise.

Reply:
Making a simplification proposal via https://github.com/WebKit/WebKit/pull/20432